pkg/controller/volume/pvcprotection/pvc_protection_controller.go

```diff
@@ -418,3 +505,63 @@ func (c *Controller) enqueuePVCs(pod *v1.Pod, deleted bool) {
 		}
 	}
 }
+
+func (c *Controller) getPV(pvc *v1.PersistentVolumeClaim) (*v1.PersistentVolume, error) {
+	pv, err := c.pvLister.Get(pvc.Spec.VolumeName)
+	if err != nil {
+		klog.V(3).Infof("unexpected error getting persistent volume %q from cache for claim %s/%s: %v", pvc.Spec.VolumeName, pvc.Namespace, pvc.Name, err)
+		pv, err = c.client.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
+	}
+	if pv == nil {
```

Comment:
do we really need to handle pv==nil here? Directly return pv, err would have the same result?

Reply:
removed this. And added a check for notfound in the caller to skip node in use checks. Added a test case also for it.